pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -56,6 +58,21 @@ func (a *AgentPXEFiles) Generate(dependencies asset.Parents) error {
 
 	a.imageReader = custom
 	a.cpuArch = agentArtifacts.CPUArch
+	// get the initrd.addrsize depending on the CPU arch
+	if a.cpuArch == arch.RpmArch(types.ArchitectureS390X) {
+		agentInitrdAddrFilePath := filepath.Join(a.tmpPath, "images", "initrd.addrsize")
+		addrsizeFile, err := isoeditor.NewInitrdAddrsizeReader(agentInitrdAddrFilePath, custom)
+		if err != nil {
+			return err
+		}
+		a.initrdAddrReader = addrsizeFile
+		// put custom to file start
+		_, err = custom.Seek(0, io.SeekStart)
```

Comment:
Due to the calculation of the initrd.img file the file pointer sits at the end of the initrd.img file and would lead to a empty file. Therefor I need to move the file pointer to the start again.

Reply:
As discussed I refactored the code.